### PR TITLE
[Headers][X86] Add a test for MMX/SSE intrinsics

### PR DESCRIPTION
**Pull request by pogo59**
Certain intrinsics map to builtins that require an immediate (literal) argument; make sure we report non-literal arguments.

This has been kicking around downstream for a while, and the recent removal of the MMX builtins caused me to notice it again.

### CODE REVIEW

File: clang/test/CodeGen/X86/mmx-sse2-shift-with-imm.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+// Don't include mm_malloc.h, it's system specific.
+#define __MM_MALLOC_H
+
+#include <x86intrin.h>
+
+__m64 check__mm_slli_pi16(__m64 m) {
+  // CHECK-LABEL: @check__mm_slli_pi16
+  // CHECK: @llvm.x86.sse2.pslli.w(<8 x i16> %{{.*}}, i32 {{.*}})
+  return _mm_slli_pi16(m, 8);
```
Comment (phoebewang) [inline clang/test/CodeGen/X86/mmx-sse2-shift-with-imm.c:14]: I found there are similar tests in mmx-builtins.c and sse2-builtins.c. Is there anything new testing here? Or not to add them?

File: clang/test/Headers/x86-intrinsics-imm.c [hunk 2]
```diff
@@ -0,0 +1,387 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-unknown -fsyntax-only \
```
Comment (phoebewang) [inline clang/test/Headers/x86-intrinsics-imm.c:1]: Move it to clang/test/CodeGen/X86?

File: clang/test/CodeGen/X86/mmx-sse2-shift-with-imm.c (hunk 1 above)
Comment (pogo59) [reply clang/test/CodeGen/X86/mmx-sse2-shift-with-imm.c:14]: You are correct, sorry for not checking!

File: clang/test/Headers/x86-intrinsics-imm.c (hunk 2 above)
Comment (pogo59) [reply clang/test/Headers/x86-intrinsics-imm.c:1]: Okay.

**Review by phoebewang — APPROVED** [review]
LGTM, thanks!

**Pull request merged**